ci-operator/step-registry/multiarch/tuning-operator/deploy/multiarch-tuning-operator-deploy-commands.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+trap 'FRC=$?; createMTOJunit; debug' EXIT TERM
+
+# Print deployments, pods, nodes for debug purpose
+function debug() {
+    if (( FRC != 0 )); then
+        echo -e "Getting deployment info...\n"
+        echo -e "oc -n $NAMESPACE get deployments -owide\n$(oc -n $NAMESPACE get deployments -owide)"
+        echo -e "Getting pod info....\n"
+        echo -e "oc -n $NAMESPACE get pods -owide\n$(oc -n $NAMESPACE get pods -owide)"
+        echo -e "Getting nodes info...\n"
+        echo -e "oc get node -owide\n$(oc get node -owide)"
+    fi
+}
+
+# Generate the Junit for MTO
+function createMTOJunit() {
+    echo "Generating the Junit for MTO"
+    filename="import-MTO"
+    testsuite="MTO"
+    if (( FRC == 0 )); then
+        cat >"${ARTIFACT_DIR}/${filename}.xml" <<EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuite name="${testsuite}" failures="0" errors="0" skipped="0" tests="1" time="1">
+  <testcase name="OCP-00001:lwan:Installing Multiarch Tuning Operator should succeed"/>
+</testsuite>
+EOF
+    else
+        cat >"${ARTIFACT_DIR}/${filename}.xml" <<EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuite name="${testsuite}" failures="1" errors="0" skipped="0" tests="1" time="1">
+  <testcase name="OCP-00001:lwan:Installing Multiarch Tuning Operator should succeed">
+    <failure message="">Installing Multiarch Tuning Operator failed</failure>
+  </testcase>
+</testsuite>
+EOF
+    fi
+}
+
+function wait_created() {
+    # wait 10 mins
+    for _ in $(seq 1 60); do
+        if oc get "${@}" | grep -v -q "No resources found"; then
+            return 0
+        fi
+        sleep 10
+    done
+    return 1
+}
+
+if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
+    echo "Setting proxy"
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+# Deploy Multiarch Tuning Operator
+NAMESPACE="openshift-multiarch-tuning-operator"
+if [[ "$MTO_OPERATOR_INSTALL_METHOD" != "catalog" && "$MTO_OPERATOR_INSTALL_METHOD" != "bundle" ]]; then
+  echo "MTO_OPERATOR_INSTALL_METHOD must be either catalog or bundle, current value is $MTO_OPERATOR_INSTALL_METHOD"
+  exit 1
+fi
+if [[ "$MTO_OPERATOR_INSTALL_METHOD" == "catalog" ]]; then
+    KUSTOMIZE_ENABLE="true"
+fi
+if [[ "$KUSTOMIZE_ENABLE" == "true" ]]; then
```

Comment:
Is this for the future case MTO_OPERATOR_INSTALL_METHOD == 'bundle' && KUSTOMIZE_ENV == 'prow-bundle'?

If so, what if we set KUSTOMIZE_ENV to "" by default and just use
```
if [[ "$MTO_OPERATOR_INSTALL_METHOD" == "catalog" ]]; then
    KUSTOMIZE_ENV="${KUSTOMIZE_ENV:-prow}"
fi
if [[ -n "$KUSTOMIZE_ENV" ]]; then
```

- When bundle + "prow-bundle" we enter the second if.
- When bundle + "", we do not enter it
- When catalog + "" we enter it as we set it to Prow in the code.
- When catalog + "any-env" we enter it and respect the requested env.

Did I cover all?

Reply:
yes, it covers all.